scripts/us_census/acs5yr/subject_tables/common/generateColMap.py

```diff
@@ -0,0 +1,94 @@
+"""Module to generate a column map to their corresponding statistical variable defintions"""
+
+import os
+import json
+import pandas as pd
+
+#TODO:Remove this import and make it from ~/data/utils/ directory when it is available
+os.system('curl https://raw.githubusercontent.com/Abilityguy/data/dcid_gen/util/statvar_dcid_generator.py -LO statvar_dcid_generator.py')
+import statvar_dcid_generator
+
+dcid_gen = statvar_dcid_generator.StatVarDcidGenerator()
+
+class generateColMapBase:
+  """module to generate a column map given a list of columns of the dataset and a JSON Spec"""
+  def __init__(self, specDict=None, columnList=None):
+    """module init """
+    self.features = specDict
+    self.columnList = columnList
+    self.columnMap = {}
+    self.makeColMap()
+
+  def makeColMap(self):
+    #for each column generate the definition of their respective statistical variable node
+    for col in self.columnList:
+      if 'ignoreColumns' in self.features and col not in self.features['ignoreColumns']:
+          self.columnMap[col] = self.column_to_statVar(col)
+    #TODO: Enable validating the generate columnMap
+    #self.valid_colMap():
+    return self.columnMap
+
+  def column_to_statVar(self, column):
+    """generates a dictionary statistical variable with all properties specified in the JSON spec for a single column"""
+    base = False
+    statVar = {}
+    partList = column.split('!!')
+
+    for part in partList:
+      #set the base for special cases like median, etc.
+      if not base and 'measurement' in self.features:
+        if part in self.features['measurement']:
+          statVar.update(self.features['measurement'][part])
+          base = True
+
+    #set the default statVar definition
+    if not base and 'measurement' in self.features and '_DEFAULT' in self.features['measurement']:
+      statVar.update(self.features['measurement']['_DEFAULT'])
+
+    #set the populationType attribute
+    statVar['populationType'] = self.get_populationType(partList)
+
+    # associate pvs to statVar
+    if partList[-1] not in self.features['measurement']:
+      if 'pvs' in self.features:
+        #p = property, k = dictionary of partList[-1] to propertyValue
+        for p, k in self.features['pvs'].items():
+          for c, v in k.items():
+            if c == partList[-1]: #last token of column name matches in property
+              statVar[p] = v
+              if 'inferredSpec' in self.features and p in self.features['inferredSpec']:
+                statVar.update(self.features['inferredSpec'][p])
+            ##if the property is somewhere in the column name
+            for part in partList:
+              if c in part:
+                statVar[p] = v
```

Comment:
This is confusing.  Why do we do exact match in certain cases and non-exact in certain others? Feels like it should be consistent.

Reply:
I have add the reason as comment with an example in the upcoming commit.